### PR TITLE
move draft utilities to HubJS

### PR DESCRIPTION
**Pull request by drewctate**
These are all the functions that are used for the site/page draft support in the app.

### CODE REVIEW

File: packages/common/src/resources/string-to-blob.ts [hunk 1]
```diff
@@ -1,17 +1,21 @@
 /**
  * Given a string, return it as a blob
+ * NOTE: USE objectToJsonBlob if you're saving a JSON resource!!!
  * NOTE: This is not currently supported in Node
  * @param {string} the string
  */
-export function stringToBlob(s: string): any {
+export function stringToBlob(
+  s: string,
+  options: BlobPropertyBag = { type: "application/octet-stream" }
```
Comment (dbouwman) [inline packages/common/src/resources/string-to-blob.ts:9]: how about we make this param `type` and we default it to `application/octet-stream`? the only other thing the Blob constructor takes in it's options is an `endings`

File: packages/sites/src/drafts/_get-json-resource.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { IHubRequestOptions } from "@esri/hub-common";
+import { getPortalUrl } from "@esri/arcgis-rest-portal";
+import { request, IRequestOptions } from "@esri/arcgis-rest-request";
+
+/**
+ * Dog-eared for rest-js
```
Comment (dbouwman) [inline packages/sites/src/drafts/_get-json-resource.ts:6]: Actual comment :)

File: packages/sites/src/drafts/_get-json-resource.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ */
+export function _getJsonResource(
+  itemId: string,
+  resourceName: string,
+  requestOptions: IRequestOptions
+) {
+  const url = `${getPortalUrl(
+    requestOptions
+  )}/content/items/${itemId}/resources/${resourceName}`;
+  const options = Object.assign({ params: { f: "json" } }, requestOptions);
```
Comment (dbouwman) [inline packages/sites/src/drafts/_get-json-resource.ts:19]: I'm not sure `f=json` is really a thing on this? i.e. this loads https://dc.mapsqa.arcgis.com/sharing/rest/content/items/a9e4560a26b848e899a0bbff890b1a4a/resources/draft-1598396441045.json

File: packages/sites/src/drafts/delete-draft.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  siteOrPageModel: IModel,
+  hubRequestOptions: IHubRequestOptions
+) {
+  const {
+    item: { id, owner }
+  } = siteOrPageModel;
+  return _getDraftResourceName(id, hubRequestOptions).then(
+    (draftResourceName: string) => {
+      if (draftResourceName) {
+        return removeItemResource({
```
Comment (dbouwman) [inline packages/sites/src/drafts/delete-draft.ts:20]: you could consider `failSafe`-ing this call

File: packages/sites/src/drafts/fetch-and-apply-draft.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { IModel, IHubRequestOptions, getProp } from "@esri/hub-common";
+import { fetchDraft } from "./fetch-draft";
+import { applyDraft } from "./apply-draft";
+
+/**
+ * Convenience function
```
Comment (dbouwman) [inline packages/sites/src/drafts/fetch-and-apply-draft.ts:6]: Actual comment

File: packages/sites/src/drafts/page-draft-include-list.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export const PAGE_DRAFT_INCLUDE_LIST = [
```
Comment (dbouwman) [inline packages/sites/src/drafts/page-draft-include-list.ts:1]: Comment

File: packages/sites/src/drafts/save-published-status.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { IModel, IHubRequestOptions, getProp } from "@esri/hub-common";
+import { getSiteItemType } from "../get-site-item-type";
+import { updateSite } from "../update-site";
+import { getPageItemType, updatePage } from "../pages";
+
+export function savePublishedStatus(
```
Comment (dbouwman) [inline packages/sites/src/drafts/save-published-status.ts:6]: comment

File: packages/sites/src/drafts/site-draft-include-list.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+export const SITE_DRAFT_INCLUDE_LIST = [
```
Comment (dbouwman) [inline packages/sites/src/drafts/site-draft-include-list.ts:1]: comment

File: packages/sites/src/drafts/types.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import { IModel } from "@esri/hub-common";
+
```
Comment (dbouwman) [inline packages/sites/src/drafts/types.ts:2]: we have a types package... I think use that?

File: packages/common/src/resources/string-to-blob.ts (hunk 1 above)
Comment (drewctate) [reply packages/common/src/resources/string-to-blob.ts:9]: Sure.

File: packages/sites/src/drafts/_get-json-resource.ts (hunk 3 above)
Comment (drewctate) [reply packages/sites/src/drafts/_get-json-resource.ts:19]: It's to tell the `request` function how to parse the response. Ref: https://github.com/Esri/arcgis-rest-js/blob/master/packages/arcgis-rest-request/src/request.ts#L370

File: packages/sites/src/drafts/delete-draft.ts (hunk 4 above)
Comment (drewctate) [reply packages/sites/src/drafts/delete-draft.ts:20]: Tell me more. Why wouldn't we want it to throw if the draft didn't get removed?

File: packages/sites/src/drafts/types.ts (hunk 9 above)
Comment (drewctate) [reply packages/sites/src/drafts/types.ts:2]: Types package is pretty sparse. Do you want me to move the types currently in `hub-common` over as part of this PR? @dbouwman

File: packages/sites/src/drafts/_get-json-resource.ts (hunk 3 above)
Comment (dbouwman) [reply packages/sites/src/drafts/_get-json-resource.ts:20]: 👍

**Pull request merged**